packages/wonder-blocks-core/util/get-element-intersection.test.js

```diff
@@ -0,0 +1,400 @@
+                    </div>
+                );
+
+                mount(nodes);
+            };
+
+            const actAndAssert = (node) => {
+                const element = (ReactDOM.findDOMNode(node): any);
+                element.getBoundingClientRect = jest.fn(() =>
+                    rect(-50, 150, -25, 125),
+                );
+
+                // Act
+                const result = getElementIntersection(element);
+
+                // Assert
+                expect(result).toMatchObject({
+                    horizontal: "after",
+                    vertical: "before",
+                });
+                done();
+            };
+
+            arrange(actAndAssert);
+        });
+
+        test("multiple scroll ancestors, stops enumeration on first before/after", (done) => {
+            // Arrange
+            // First, let's fake out some scroll parent.
+            // Flow doesn't like jest mocks $FlowFixMe
+            enumerateScrollAncestors.mockImplementation(() => [
+                {getBoundingClientRect: jest.fn(() => rect(0, 100, 100, 0))},
+                {getBoundingClientRect: jest.fn(() => rect(50, 200, 250, 0))},
+                null,
+            ]);
+            const arrange = (actAndAssert) => {
+                const nodes = (
+                    <div>
+                        <div ref={actAndAssert}>Test</div>
+                    </div>
+                );
+
+                mount(nodes);
+            };
+
+            const actAndAssert = (node) => {
+                const element = (ReactDOM.findDOMNode(node): any);
+                element.getBoundingClientRect = jest.fn(() =>
+                    rect(0, 25, 25, 0),
+                );
+
+                // Act
+                const underTest = () => getElementIntersection(element);
+
+                // Assert
+                expect(underTest).not.toThrow();
+                done();
+            };
+
+            arrange(actAndAssert);
+        });
+    });
+
+    describe("when bounds element is not falsy", () => {
+        test("does not enumerate scroll ancestors", (done) => {
+            // Arrange
+            const fakeParentElement = ({
+                getBoundingClientRect: jest.fn(() => rect(0, 100, 100, 0)),
+            }: any);
+            const arrange = (actAndAssert) => {
+                const nodes = (
+                    <div>
+                        <div ref={actAndAssert}>Test</div>
+                    </div>
+                );
+
+                mount(nodes);
+            };
+
+            const actAndAssert = (node) => {
+                const element = (ReactDOM.findDOMNode(node): any);
+
+                // Act
+                getElementIntersection(element, fakeParentElement);
+
+                // Assert
+                expect(enumerateScrollAncestors).not.toHaveBeenCalledWith(
+                    element,
+                );
+                done();
+            };
+
+            arrange(actAndAssert);
```

Comment:
You have a number of similar tests, it might be worth making a helper, e.g.
```
testHelper(arrayOfScrollAncestors, underTestBounds, expectedResult);
```

Reply:
I don't like linking tests that way if I can help it. It makes test maintenance much harder. Generally, keeping test implementations isolated from one another keeps things clear and unambiguous.